src/battle_transition.c

```diff
@@ -0,0 +1,2513 @@
+	sMugshotPal_Drake,
+	sMugshotPal_Steven
+};
+
+static const u16 * const sPlayerMugshotsPals[2] =
+{
+    sMugshotPal_Brendan,
+    sMugshotPal_May
+};
+
+static const u16 sUnusedTrainerPalette[] = INCBIN_U16("graphics/battle_transitions/unused_trainer.gbapal");
+static const struct SpritePalette sSpritePalette_UnusedTrainer =
+{
+    sUnusedTrainerPalette, 0x100A
+};
+
+static const u16 sBigPokeball_Tilemap[] = INCBIN_U16("graphics/battle_transitions/big_pokeball_map.bin");
+static const u16 sMugshotsTilemap[] = INCBIN_U16("graphics/battle_transitions/elite_four_bg_map.bin");
+
+// actual code starts here
+
+void sub_811AABC(u8 transitionID)
+{
+    gMain.callback2 = sub_8054398;
+    LaunchBattleTransitionTask(transitionID);
+}
+
+void sub_811AAD8(u8 transitionID)
+{
+    LaunchBattleTransitionTask(transitionID);
+}
+
+#define tState          data[0]
+#define tTransitionID   data[1]
+#define tTransitionDone data[15]
+
+bool8 IsBattleTransitionDone(void)
+{
+    u8 taskID = FindTaskIdByFunc(Task_BattleTransitionMain);
+    if (gTasks[taskID].tTransitionDone)
+    {
+        DestroyTask(taskID);
+        return TRUE;
+    }
+    else
+        return FALSE;
+}
+
+static void LaunchBattleTransitionTask(u8 transitionID)
+{
+    u8 taskID = CreateTask(Task_BattleTransitionMain, 2);
+    gTasks[taskID].tTransitionID = transitionID;
+}
+
+static void Task_BattleTransitionMain(u8 taskID)
+{
+    while (sMainTransitionPhases[gTasks[taskID].tState](&gTasks[taskID]));
+}
+
+static bool8 Transition_Phase1(struct Task* task)
+{
+    sub_807DE10();
+    CpuSet(gPlttBufferFaded, gPlttBufferUnfaded, 0x4000100);
+    if (sPhase1_Tasks[task->tTransitionID] != NULL)
+    {
+        CreateTask(sPhase1_Tasks[task->tTransitionID], 4);
+        task->tState++;
+        return FALSE;
+    }
+    else
+    {
+        task->tState = 2;
+        return TRUE;
+    }
+}
+
+static bool8 Transition_WaitForPhase1(struct Task* task)
+{
+    if (FindTaskIdByFunc(sPhase1_Tasks[task->tTransitionID]) == 0xFF)
+    {
+        task->tState++;
+        return TRUE;
+    }
+    else
+        return FALSE;
+}
+
+static bool8 Transition_Phase2(struct Task* task)
+{
+    CreateTask(sPhase2_Tasks[task->tTransitionID], 0);
+    task->tState++;
+    return FALSE;
+}
+
+static bool8 Transition_WaitForPhase2(struct Task* task)
+{
+    task->tTransitionDone = 0;
+    if (FindTaskIdByFunc(sPhase2_Tasks[task->tTransitionID]) == 0xFF)
+        task->tTransitionDone = 1;
+    return FALSE;
+}
+
+static void Phase1Task_TransitionAll(u8 taskID)
+{
+    if (gTasks[taskID].tState == 0)
+    {
+        gTasks[taskID].tState++;
+        CreatePhase1Task(0, 0, 3, 2, 2);
+    }
+    else if (sub_811D52C())
+        DestroyTask(taskID);
+}
+
+static void Phase2Task_Transition_Blur(u8 taskID)
+{
+    while (sPhase2_Transition_Blur_Funcs[gTasks[taskID].tState](&gTasks[taskID]));
+}
+
+static bool8 Phase2_Transition_Blur_Func1(struct Task* task)
+{
+    REG_MOSAIC = 0;
+    REG_BG1CNT |= 0x40;
```

Comment:
The video register flags are in include/gba/io_reg.h. Mosaic is that pixelated blocky effect.